recipes/MOABB/MotorImagery_decoding/BNCI2014001/train.yaml

```diff
@@ -0,0 +1,144 @@
+###################################################
+#
+# Author
+# ------
+# Davide Borra, 2021
+###################################################
+seed: 2602
+__set_seed: !apply:torch.manual_seed [!ref <seed>]
+
+# DIRECTORIES
+data_folder: !PLACEHOLDER  #'/path/to/MOABB_BNCI2014001'
+output_folder: '/home/prometheus/Documents/results/MOABB_BNCI2014001' #!ref results/MOABB_BNCI2014001/<seed>
+
+# DATASET HPARS
+C: 22
+# Target sampling rate while downsampling
+sf: 128
+# tmin, tmax respect to stimulus onset to define EEG epochs
+tmin: 0.5
+tmax: 2.5
+# fmin, fmax band-pass filtering
+fmin: 4
+fmax: 40
+# Time samples=int((tmax-tmin)*sf)
+T: 256
+
+# TRAINING HPARS
+nfolds: 5
+valid_ratio: 0.2
+number_of_epochs: 500 # number of training epochs
+max_increase_epochs: 50  # early stop when after consecutive "max_increase_epochs" epochs there is no improvement
+num_warmup_epochs: 5  # start checking for early stopping after first "num_warmup_epochs"
+target_valid_metric: 'f1'  # target metric to track while performing early stopping
+direction: 'max'  # direction to optimize the target metric
+batch_size: 32
+lr: 0.001
+loss: !name:torch.nn.functional.nll_loss
+class_weight: [1, 1, 1, 1]  # class weight to address class imbalance
+optimizer: !name:torch.optim.Adam
+  lr: !ref <lr>
+epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounterWithStopper  # epoch counter and early stopper
+  limit: !ref <number_of_epochs>
+  limit_to_stop: !ref <max_increase_epochs>
+  limit_warmup: !ref <num_warmup_epochs>
+  direction: !ref <direction>
+
+
+# EEGNET HPARS
+k0: 8  # number of temporal filters in the temporal conv. layer
+d1: 2  # depth multiplier for the spatial depthwise conv. layer
+k3: !ref <k0> * <d1>  # number of spatial depthwise filters
+f0: 33  # length of temporal filters in the temporal conv. layer
+f2: 17  # length of temporal filters in the temporal separable conv. layer
+fp0: 4  # temporal pool size and stride of the first pooling layer
+fp1: 8  # temporal pool size and stride of the second pooling layer
+fptot: !ref <fp0> * <fp1>  # overall pool size
+max_norm_1: 1.  # kernel max-norm constaint of the spatial depthwise conv. layer
+max_norm_out: 0.25  # kernel max-norm constaint of the dense layer
+p_drop: 0.5  # dropout rate
+padding_mode: 'constant'
+
+layer0: !new:speechbrain.nnet.CNN.Conv2d
```

Comment:
An alternative here would be to define the model in lobes (e.g. speechbrain/lobes/EEGNET.py). However, defining it every time doesn't look that bad to me in the end.